src/main.rs

```diff
@@ -0,0 +1,219 @@
+                    alive_neighbors_count(row as i32, col as i32, self) >3 { // underpopulation or overpopulation
+                        cells_copy[row][col] = Cell::Dead; // but modify the copy
+                    }
+                } else {
+
+                    if alive_neighbors_count(row as i32, col as i32, self) == 3 { // reproduction
+                        cells_copy[row][col] = Cell::Alive
+                    }
+                }
+            }
+        }
+        self.cells = cells_copy; // finally we modify the state all at once
+    }
+
+    pub fn draw_cells(&self){
+        for row in 0..self.cells.len() {
+            for col in 0..self.cells[row].len() {
+
+                if self.cells[row][col].is_alive() { // alive cells are black rectangles
+                    draw_rectangle((col as f32) * CELL_SIZE, (row as f32) *CELL_SIZE, CELL_SIZE , CELL_SIZE, BLACK);
+
+                } else { // dead cells are white rectangles
+                    draw_rectangle((col as f32) * CELL_SIZE, (row as f32) *CELL_SIZE, CELL_SIZE , CELL_SIZE, WHITE);
+                }
+            }
+        }
+    }
+
+    pub fn draw_grid(&self) {
+        let mut x: f32 = 0.0;
+        let mut y: f32 = 0.0;
+
+        while x + CELL_SIZE <= screen_width() {
+            draw_line(x, 0.0, x, screen_height(), 1.0, LIGHTGRAY);
+            x += CELL_SIZE;
+        }
+        while y + CELL_SIZE <= screen_height() {
+            draw_line(0.0, y, screen_width(), y, 1.0, LIGHTGRAY);
+            y += CELL_SIZE;
+        }
+    }
+}
+
+pub fn alive_neighbors_count(row: i32, col: i32, grid: &mut Grid) -> i32 {
+    let mut count = 0;
+    let neighbors: [(i32, i32); 8] = [(0,-1) , (-1,0), (0,1), (1,0), (1,-1), (-1,1), (-1,-1), (1,1)];
+
+    for (x, y) in neighbors.into_iter() {
+
+        if (row + y < 0) || (row + y >= grid.height) || (col + x < 0) || (col + x >= grid.width) {
+            continue; // to avoid non valid indexes since the universe is finite
+
+        } else {
+            if grid.cells[(row + y) as usize][(col + x) as usize].is_alive(){
+                count += 1;
+            }
+        }
+    }
+    count
+}
+
+#[macroquad::main("Game of Life")]
+async fn main() {
+    let w = screen_width() as usize;
+    let h = screen_height() as usize;
+    let x_quantity = w as f32 / CELL_SIZE; // number of cells
+    let y_quantity = h as f32 / CELL_SIZE;
+
+    let mut universe = Grid {
+        width: x_quantity as i32,
+        height: y_quantity as i32,
+        cells: get_default_pattern(x_quantity,y_quantity)
+    };
+    clear_background(WHITE);
+    universe.draw_cells(); // to see initial state
+    universe.draw_grid();
+    next_frame().await;
+    thread::sleep(time::Duration::from_secs(1));
+
+    loop {
+        universe.tick();
+        universe.draw_grid(); 
+        next_frame().await; // // submit our render calls to our screen
+        thread::sleep(time::Duration::from_secs(1));
```

Comment:
These would not work with WASM because of the usage of `thread::sleep` this is a good way to control the frames per second (FPS) running the app normally but fails to run in the browser. There is an issue on the `macroquad` lib asking how to achieve this compiling with WASM https://github.com/not-fl3/macroquad/issues/380 (TLDR: there's a workaround but it's not supported on the lib). Right now it would be okay to allow the app runs freely without limiting the FPS. You can add `#[cfg(not(target_arch = "wasm32"))]` above both `sleep` calls to tell the compiler to execute that lines unless it's compiling for WASM.